src/account/accountsSelectors.js

```diff
@@ -88,6 +108,25 @@ export const getAccount = (state: PerAccountState): Account => {
   return accounts[0];
 };
 
+/**
+ * Whether the given identity matches the active account.
+ *
+ * Gives false if there is no active account.
+ */
+export const getIsActiveAccount: GlobalSelector<(Identity) => boolean> = createSelector(
```

Comment:
This doesn't benefit from memoization (i.e. from `createSelector`) — the callback we pass to `createSelector` is not doing any work, just immediately returning a function expression.

Removing that will simplify things. Then I think let's simplify further by uncurrying, so that this is just a function on two arguments rather than a function on one argument that returns a function on the other argument.

Reply:
Makes sense. I think I was looking at this other selector when writing this one:

```js
/**
 * All known accounts, indexed by identity.
 */
export const getAccountsByIdentity: GlobalSelector<(Identity) => Account | void> = createSelector(
  getAccounts,
  accounts => {
    const map = new Map(
      accounts.map(account => [keyOfIdentity(identityOfAccount(account)), account]),
    );
    return identity => map.get(keyOfIdentity(identity));
  },
);
```

That one is "curried" too, right? I wonder if there's a way to simplify that one, while still enforcing that callers pass the the right kind of thing to identify the account (whether an `Identity` or a `keyOfIdentity` string). Anyway, not worth spending too much time on.

Reply:
That one has the type of a curried function, in that it's of the form `A => (B => C)` (whereas the uncurried counterpart of that type would be `(A, B) => C`.

But the implementation has more going on. An implementation that's purely curried without more happening would look like:
```js
export const getAccountsByIdentity: GlobalState => ((Identity) => Account | void) = state => identity => {
  const accounts = getAccounts(state);
  const map = new Map(
    accounts.map(account => [keyOfIdentity(identityOfAccount(account)), account]),
  );
  return map.get(keyOfIdentity(identity));
};
```
The actual implementation using `createSelector`, though, does a cache lookup before the `new Map` step. Because the `new Map` represents doing some work (constructing a new data structure, vs. just looking up properties in existing objects and doing simple boolean tests), that caching is potentially useful.